src/test/compile-fail/coherence-impls-builtin.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::marker::Send;
+
+enum TestE {
+  A
+}
+
+struct MyType;
+
+unsafe impl Send for TestE {}
+unsafe impl Send for MyType {}
+unsafe impl Send for (MyType, MyType) {}
+//~^ ERROR builtin traits can only be implemented on `structs` or `enums`
```

Comment:
Let's add a few more impls for good measure. For example `&MyType`, `[MyType]`, and `&[MyType]`.